### PR TITLE
Implement triggering of powerups

### PR DESCRIPTION
**Pull request by malcx95**
Solves #102

### CODE REVIEW

**Review by rorro — APPROVED** [review]

File: libplen/src/gamestate.rs [hunk 1]
```diff
@@ -120,7 +120,7 @@ impl GameState {
                     let hit_radius = PLANE_SIZE + POWERUP_RADIUS;
                     if (powerup.position - player.position).norm() < hit_radius as f32 {
                         // Apply the powerup
```
Comment (TheZoq2) [inline libplen/src/gamestate.rs]: Outdated comment

File: libplen/src/messages.rs [hunk 2]
```diff
@@ -93,6 +93,6 @@ pub enum ServerMessage {
 
 #[derive(Serialize, Deserialize)]
 pub enum ClientMessage {
-    Input { x_input: f32, y_input: f32, shooting: bool },
+    Input { x_input: f32, y_input: f32, shooting: bool, firing_powerup: bool },
```
Comment (TheZoq2) [inline libplen/src/messages.rs]: Maybe `using` or `activating` would be a more descriptive name

File: libplen/src/player.rs [hunk 3]
```diff
@@ -322,14 +324,32 @@ impl Player {
...
+    }
+
+    pub fn trigger_powerup_if_available(&mut self) {
+        match self.available_powerup {
+            None => (),
+            Some(powerup) => {
+                self.apply_powerup(powerup);
+                self.available_powerup = None;
+            }
+        }
```
Comment (TheZoq2) [inline libplen/src/player.rs]: Use `if let`

File: src/server.rs [hunk 4]
```diff
@@ -175,14 +175,16 @@ impl Server {
             let mut player_input_x = 0.0;
             let mut player_input_y = 0.0;
             let mut player_shooting = false;
+            let mut player_firing_powerup = false;
```
Comment (TheZoq2) [inline src/server.rs]: Same as above, `player_activating_powerup` is probably a better name

File: src/server.rs [hunk 5]
```diff
@@ -175,14 +175,16 @@ impl Server {
...
 
             // TODO: Use a real loop
             while let Some(message) = client.next() {
                 match message {
-                    ClientMessage::Input{ x_input, y_input, shooting } => {
+                    ClientMessage::Input{ x_input, y_input, shooting, firing_powerup } => {
```
Comment (TheZoq2) [inline src/server.rs]: Same here

**Review by TheZoq2 — APPROVED** [review]

**Pull request merged**